File: CodingSamples/DotNet/Inheritance/AbstractClassTest/Banking/CurrentAccount.cs
namespace Banking;

//class defined with sealed modifier cannot be used as a base class,
//it can only be used for activation
sealed class CurrentAccount : Account
{
    public override void Deposit(double amount)
    {
        Balance += amount;
    }

    public override void Withdraw(double amount)
    {
        Balance -= amount;
    }
}

File: CodingSamples/Java/Inheritance/AbstractClassTest/src/CurrentAccount.java
package banking;

//a class defined with final modifier cannot be used as a super-class
final class CurrentAccount extends Account {

	public void deposit(double amount) {
		if(balance < 0)
			amount *= 0.9;
		balance += amount;
	}

	public void withdraw(double amount) throws InsufficientFundsException {
		balance -= amount;
	}
	
}

